archive/frames/serializers.py

```diff
@@ -55,32 +55,37 @@ class FrameSerializer(serializers.ModelSerializer):
     url = serializers.CharField(read_only=True, help_text='File download URL associated with most recent version')
     filename = serializers.CharField(read_only=True, help_text='Full filename')
     area = PolygonField(allow_null=True, help_text='GeoJSON area that this frame covers')
-    DAY_OBS = serializers.DateField(input_formats=['iso-8601', '%Y%m%d'], help_text='Observation day in %Y%m%d format')
-    related_frames = serializers.PrimaryKeyRelatedField(
-        many=True,
-        queryset=Frame.objects.exclude(DATE_OBS=None),
-        required=False,
+    observation_day = serializers.DateField(input_formats=['iso-8601', '%Y%m%d'], help_text='Observation day in %Y%m%d format')
+    headers = serializers.JSONField(required=True, write_only=True)
+    related_frame_filenames = serializers.ListField(
+        child=serializers.CharField(), required=True, write_only=True,
         style={'base_template': 'input.html'},
         help_text='Set of related frames for this file'
     )
 
     class Meta:
         model = Frame
         fields = (
-            'id', 'basename', 'area', 'related_frames', 'version_set',
-            'filename', 'url', 'RLEVEL', 'DAY_OBS', 'DATE_OBS', 'PROPID',
-            'INSTRUME', 'OBJECT', 'SITEID', 'TELID', 'EXPTIME', 'FILTER',
-            'L1PUBDAT', 'OBSTYPE', 'BLKUID', 'REQNUM',
+            'id', 'basename', 'area', 'related_frames', 'version_set', 'headers',
```

Comment:
Do we need to provide the old field names in the response as well? Clients that parse the response from the GET frames endpoint might break if we don't.

Reply:
hmm yeah I added back the model fields but not these. I guess it might be appropriate to send these as well for a short period of time while telling end users to update their code, in case they are using them (like the Tom Toolkit and older toms...)